packages/monosize-storage-azure/src/createTableClient.test.mts

```diff
@@ -0,0 +1,76 @@
+import { describe, expect, it, vitest } from 'vitest';
+import { AzureNamedKeyCredential, TableClient } from '@azure/data-tables';
+import { AzurePipelinesCredential } from '@azure/identity';
+import { createTableClient } from './createTableClient.mjs';
+import { beforeEach } from 'node:test';
+
+vitest.mock('@azure/data-tables', () => {
+  return {
+    AzureNamedKeyCredential: vitest.fn(),
+    TableClient: vitest.fn().mockImplementation(() => {
+      return {
+        createTable: vitest.fn(),
+        deleteTable: vitest.fn(),
+      };
+    }),
+  };
+});
+
+vitest.mock('@azure/identity', () => {
+  return {
+    AzurePipelinesCredential: vitest.fn(),
+  };
+});
+
+describe('createTableClient', () => {
+  beforeEach(() => {
+    vitest.resetAllMocks();
+    vitest.unstubAllEnvs();
+  });
+
+  it('should create TableClient with AzureNamedKeyCredential', () => {
+    vitest.stubEnv('BUNDLESIZE_ACCOUNT_NAME', 'test-account-name');
+    vitest.stubEnv('BUNDLESIZE_ACCOUNT_KEY', 'test-account-key');
+
+    const authType = 'AzureNamedKeyCredential';
+    const tableName = 'test-table';
+    createTableClient({ authType, tableName });
+
+    expect(AzureNamedKeyCredential).toHaveBeenCalledWith(
+      process.env['BUNDLESIZE_ACCOUNT_NAME'] as string,
+      process.env['BUNDLESIZE_ACCOUNT_KEY'] as string,
+    );
+
+    expect(TableClient).toHaveBeenCalledWith(
+      'https://test-account-name.table.core.windows.net',
+      tableName,
+      expect.any(AzureNamedKeyCredential),
+    );
+  });
+
+  it('should create TableClient with AzurePipelinesCredential', () => {
```

Comment:
let's add one more to catch typos - our throw error escape hatch

Reply:
Added!